### PR TITLE
New formula: docker-credential-helper-ecr

### PR DESCRIPTION
**Pull request by singingwolfboy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Follow-up from #18914. Repository: https://github.com/awslabs/amazon-ecr-credential-helper

### CODE REVIEW

**Comment by singingwolfboy** [comment]
@ilovezfs @DomT4 Do either of you have any suggestions on how to fix this problem? I'm happy to change the description if you can suggest a better one. Alternatively, is there a way to tell the linter to ignore this problem?

**Comment by DomT4** [comment]
Does something like `EC2 Container Registry Docker Credential Helper` make relative sense?

**Comment by singingwolfboy** [comment]
That's a bit hard to parse, for me at least. Lots of buzzwords all slammed together -- it's hard to tell where one concept ends and the next begins. What's a "Registry Docker Credential", and how does it help an "EC2 Container"? :smile:

**Comment by samuelkarp** [comment]
@singingwolfboy How about "Docker Credential Helper for Amazon ECR"?

**Comment by DomT4** [comment]
That seems a semi-reasonable suggestion?

**Comment by samuelkarp** [comment]
Alternate suggestions: "Amazon ECR Docker Credential Helper", "Amazon ECR Credential Helper for Docker"

**Comment by singingwolfboy** [comment]
Alright, the description is now "Docker Credential Helper for Amazon ECR". Between that, and removing the `head` line, the build is now green. Is there anything else I need to do, or can this be merged?

**Review by DomT4 — APPROVED** [review]
Seems reasonable enough to me but will need Joe to look it over.

**Comment by singingwolfboy** [comment]
Is there anything preventing this formula from being merged?

File: Formula/docker-credential-helper-ecr.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class DockerCredentialHelperEcr < Formula
+  desc "Docker Credential Helper for Amazon ECR"
```
Comment (ilovezfs) [inline Formula/docker-credential-helper-ecr.rb]: The description shouldn't really just be the name.

File: Formula/docker-credential-helper-ecr.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    cd dir do
+      system "make", "build"
+      bin.install "bin/local/docker-credential-ecr-login"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    run_output = shell_output("#{bin}/docker-credential-ecr-login", 1)
```
Comment (ilovezfs) [inline Formula/docker-credential-helper-ecr.rb]: `output =` is the standard variable name for this

**Review by ilovezfs — CHANGES_REQUESTED** [review]
I'd like to see a test that exercises and verifies some of the functionality.

File: Formula/docker-credential-helper-ecr.rb (hunk 1 above)
Comment (singingwolfboy) [reply Formula/docker-credential-helper-ecr.rb]: @ilovezfs I'm open to suggestions, but considering the character limit that `brew audit` enforces, there isn't any other good option.

**Comment by singingwolfboy** [comment]
pinging @samuelkarp

**Review by fxcoudert — APPROVED** [review]
Given how long this has waited, I'd say merge as is, but please improve the test in a later PR.

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
Thanks @singingwolfboy and @samuelkarp for your contribution to Homebrew! Sorry for the slow review, this is now merged.

Please consider opening a PR in the future with an improved test. Possibly a test that tries to connect to the remote service with dumb credentials, and matches the failure / error message.
